src/Raven.Server/Documents/TimeSeries/TimeSeriesPolicyRunner.cs

```diff
@@ -0,0 +1,683 @@
+                var table = context.Transaction.InnerTransaction.OpenTable(RollUpSchema, TimeSeriesRollUps);
+
+                var toDelete = new List<Slice>(tss.Stats.GetTimeSeriesNameByPolicy(context, _collection, _policy, 0, BatchSize));
+
+                var request = new TimeSeriesStorage.DeletionRangeRequest
+                {
+                    Collection = _collection.Name,
+                    From = DateTime.MinValue,
+                    To = DateTime.MaxValue,
+                };
+
+                foreach (var key in toDelete)
+                {
+                    SplitKey(key, out var docId, out var name);
+                    request.DocumentId = docId;
+                    request.Name = name;
+                    
+                    tss.RemoveTimestampRange(context, request);
+
+                    table.DeleteByKey(key);
+
+                    Deleted++;
+                }
+
+                return Deleted;
+            }
+
+            public override TransactionOperationsMerger.IReplayableCommandDto<TransactionOperationsMerger.MergedTransactionCommand> ToDto(JsonOperationContext context)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        internal class AddedNewRollupPoliciesCommand : TransactionOperationsMerger.MergedTransactionCommand
+        {
+            public const int BatchSize = 1024;
+            private readonly CollectionName _collection;
+            private readonly TimeSeriesPolicy _from;
+            private readonly TimeSeriesPolicy _to;
+            private readonly int _skip;
+
+            public int Marked;
+
+            public AddedNewRollupPoliciesCommand(CollectionName collection, TimeSeriesPolicy from, TimeSeriesPolicy to, int skip)
+            {
+                _collection = collection;
+                _from = from;
+                _to = to;
+                _skip = skip;
+            }
+            protected override long ExecuteCmd(DocumentsOperationContext context)
+            {
+                var tss = context.DocumentDatabase.DocumentsStorage.TimeSeriesStorage;
+                RollUpSchema.Create(context.Transaction.InnerTransaction, TimeSeriesRollUps, 16);
+                var table = context.Transaction.InnerTransaction.OpenTable(RollUpSchema, TimeSeriesRollUps);
+                foreach (var key in tss.Stats.GetTimeSeriesNameByPolicy(context, _collection, _from.Name, _skip, BatchSize))
+                {
+                    using (table.Allocate(out var tvb))
+                    using (Slice.From(context.Allocator,_collection.Name, ByteStringType.Immutable,out var collectionSlice))
+                    using (Slice.From(context.Allocator, _to.Name, ByteStringType.Immutable, out var policyToApply))
+                    using (Slice.From(context.Allocator, string.Empty, ByteStringType.Immutable, out var changeVectorSlice))
+                    {
+                        tvb.Add(key);
```

Comment:
Are we sure we want code like this here instead of e.g. `TimeSeriesStorage.Rollups.MethodName()` ?

Reply:
Not sure I'm following.

Reply:
What I've meant is that we are using xxxStorage classes for storage interaction and here you are inserting directly into table without using TimeSeriesStorage.Rollups for example

Reply:
Okay, I will move it to dedicated class.